### PR TITLE
fs/vfs: remove template filep to optimize dup operation 

### PR DESCRIPTION
**Pull request by Donny9**
## Summary
fs/vfs: remove template filep to optimize dup operation
## Impact
optimize dup code
## Testing
Vela

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



**No, the provided PR information does not meet the NuttX requirements.**

**Missing Information:**

* **Summary:**
    * Lacks a clear explanation of why the change is necessary.
    * Does not specify the functional part of the code being changed (e.g., specific functions or data structures).
    * Missing a detailed description of how the change works.
* **Impact:**
    * While it mentions "optimize dup code," it lacks specifics about the impact on users, build process, hardware, documentation, security, and compatibility.
* **Testing:**
    * "Vela" is insufficient information. 
    * Needs to specify:
        * Build Host OS, CPU, compiler details.
        * Target architecture, board, and configuration.
    * Missing testing logs before and after the change.

**To meet the NuttX requirements, the PR needs to provide comprehensive information in all sections, including specific details and relevant logs.**

**Comment by Donny9** [comment]
> [**[Experimental Bot, please feedback here]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)
> 
> **No, the provided PR information does not meet the NuttX requirements.**
> 
> **Missing Information:**
> 
> * **Summary:**
>   
>   * Lacks a clear explanation of why the change is necessary.
>   * Does not specify the functional part of the code being changed (e.g., specific functions or data structures).
>   * Missing a detailed description of how the change works.
> * **Impact:**
>   
>   * While it mentions "optimize dup code," it lacks specifics about the impact on users, build process, hardware, documentation, security, and compatibility.
> * **Testing:**
>   
>   * "Vela" is insufficient information.
>   * Needs to specify:
>     
>     * Build Host OS, CPU, compiler details.
>     * Target architecture, board, and configuration.
>   * Missing testing logs before and after the change.
> 
> **To meet the NuttX requirements, the PR needs to provide comprehensive information in all sections, including specific details and relevant logs.**

The purpose of this PR is to streamline the dup process.

Avoid using the entity filep2, which enables us to remove the memset and close operations, thereby accelerating the dup process.
Initially, find a free file descriptor (fd) from the file_list to occupy the position and assign it to the pointer filep2. Then, directly invoke file_dup3 to complete the dup operation.

**Review by jasonbu — APPROVED** [review]

File: fs/vfs/fs_dup.c [hunk 1]
```diff
@@ -52,27 +52,50 @@
...
-  if (fd2 < 0)
+  ret = fs_getfilep(fd2, &filep2);
+#ifdef CONFIG_FDSAN
+  f_tag_fdsan = filep2->f_tag_fdsan;
+#endif
+
+#ifdef CONFIG_FDCHECK
+  f_tag_fdcheck = filep2->f_tag_fdcheck;
+#endif
+  DEBUGASSERT(ret >= 0);
```
Comment (pkarashchenko) [inline fs/vfs/fs_dup.c:80]: Better to check this after `fs_getfilep` return before dereferencing `NULL` pointers.

Comment (Donny9) [reply fs/vfs/fs_dup.c:80]: > Better to check this after `fs_getfilep` return before dereferencing `NULL` pointers.

@pkarashchenko  We have already determined whether file_allocate can return correctly before this, which should be sufficient, don't you think?

**Review by xiaoxiang781216 — APPROVED** [review]

File: fs/vfs/fs_dup.c (hunk 1 above)
Comment (xiaoxiang781216) [reply fs/vfs/fs_dup.c:80]: fs_getfilep is simple to add the reference count, which should never fail, so DEBUGASSERT is enough here.

**Pull request merged**